### PR TITLE
Add Poisson solver executable

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

- [x] Depends on #1090
- [x] Refactor initializers
- [x] Use observers
- [x] Test the boundary conditions
- [x] Test all initializers
- [x] Depends on #1192
- [x] Depends on #1194
- [x] Depends on #1205 
- [x] Depends on #1206 
- [x] Depends on #1207 

### Types of changes:

- [x] New feature

### Component:

- [x] Code

### Code review checklist

- [x] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [x] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [x] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Elliptic/DiscontinuousGalerkin/DgElementArray.hpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+template <observers::TypeOfObservation TypeOfObservation>
+struct RegisterWithObservers;
+}  // namespace Actions
+}  // namespace observers
+/// \endcond
+
+namespace Elliptic {
+
+template <class Metavariables, class ActionList>
+struct DgElementArray {
```
Comment (nilsvu) [inline src/Elliptic/DiscontinuousGalerkin/DgElementArray.hpp]: Note to self: Add comment that this is the same as the evolution code right now, but may diverge in the future (e.g. for multigrid)

File: src/Elliptic/Executables/Poisson/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+set(LIBS_TO_LINK
```
Comment (nilsdeppe) [inline src/Elliptic/Executables/Poisson/CMakeLists.txt:4]: The indentation in this file is inconsistent with a lot of our CMake code. We've been doing 2 space indentation in them. I realize this probably came from another file, but we should try moving to being more consistent on this :)

**Review by nilsdeppe — COMMENTED** [review]
I have one minor stylistic comment. I'm happy for you to squash.

**Comment by nilsdeppe** [comment]
Can now be rebased on develop

**Comment by nilsvu** [comment]
done @sxs-collaboration/spectre-core-devs

**Review by nilsdeppe — DISMISSED** [review]

**Review by wthrowe — DISMISSED** [review]

File: tests/InputFiles/Poisson/Input3D.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+DomainCreator:
+  Brick:
+    LowerBound: [0, 0, 0]
+    UpperBound: [3.141592653589793, 3.141592653589793, 3.141592653589793]
+    IsPeriodicIn: [false, false, false]
+    InitialRefinement: [1, 1, 1]
+    InitialGridPoints: [2, 2, 2]
+
+NumericalFluxParams:
+    PenaltyParameter: 5.729577951308232
```
Comment (kidder) [inline tests/InputFiles/Poisson/Input3D.yaml]: where does this value come from?

**Review by kidder — DISMISSED** [review]

**Review by nilsdeppe — DISMISSED** [review]

File: tests/InputFiles/Poisson/Input3D.yaml (hunk 3 above)
Comment (wthrowe) [reply tests/InputFiles/Poisson/Input3D.yaml]: It's p^2/h = 18/pi for 1D and 2D, but p is different here.
@nilsleiffischer Should the 3D case be 8/pi instead?

Might be worth a comment in either case.

Comment (nilsvu) [reply tests/InputFiles/Poisson/Input3D.yaml]: Sure I added a comment. This is also documented in `Poisson::FirstOrderInternalPenaltyFlux`. Corrected this for p=2 in the 3D file.

**Review by kidder — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Pull request merged**